source/dofs/dof_tools_constraints.cc

```diff
@@ -1873,167 +1873,226 @@ namespace DoFTools
 
           // loop over all dofs on face 2 and constrain them against the ones on face 1
           for (unsigned int i=0; i<dofs_per_face; ++i)
-            if (!constraint_matrix.is_constrained(dofs_2[i]))
-              if ((component_mask.n_selected_components(fe.n_components())
-                   == fe.n_components())
-                  ||
-                  component_mask[fe.face_system_to_component_index(i).first])
-                {
-                  // as mentioned in the comment above this function, we need
-                  // to be careful about treating identity constraints differently.
-                  // consequently, find out whether this dof 'i' will be
-                  // identity constrained
-                  //
-                  // to check whether this is the case, first see whether there are
-                  // any weights other than 0 and 1, then in a first stage make sure
-                  // that if so there is only one weight equal to 1
-                  //
-                  // afterwards do the same for constraints of type dof1=-dof2
-                  bool is_identity_constrained = true;
-                  const double eps = 1.e-13;
+            if ((component_mask.n_selected_components(fe.n_components())
+                 == fe.n_components())
+                ||
+                component_mask[fe.face_system_to_component_index(i).first])
+              {
+                // as mentioned in the comment above this function, we need
+                // to be careful about treating identity constraints differently.
+                // consequently, find out whether this dof 'i' will be
+                // identity constrained
+                //
+                // to check whether this is the case, first see whether there are
+                // any weights other than 0 and 1, then in a first stage make sure
+                // that if so there is only one weight equal to 1
+                //
+                // afterwards do the same for constraints of type dof1=-dof2
+                bool is_identity_constrained = true;
+                const double eps = 1.e-13;
+                for (unsigned int jj=0; jj<dofs_per_face; ++jj)
+                  if (((std::abs(transformation(i,jj)) < eps) ||
+                       (std::abs(transformation(i,jj)-1) < eps)) == false)
```

Comment:
can you do ``1.``? The ``== false`` is also more complicated than I would have written it.